app/api/resources/db_connections.rb

```diff
@@ -0,0 +1,56 @@
+module Resources
+  class DbConnections < Grape::API
+    
+    helpers do
+
+      def encrypt(password)
+        key = request.headers['Encryption-Key']
+        error!('No AES key sent', 401) unless key
+
+        aes_key = Base64.strict_decode64(key)
+        cipher = OpenSSL::Cipher::AES.new(256, :CBC)
```

Comment:
I think with the changes on the PBKDF, you need to change the key length to 128 bit, since PBKDF2 generates only 128b.